mlflow/tracing/provider.py

```diff
@@ -215,15 +215,26 @@ def f():
 
     @functools.wraps(f)
     def wrapper(*args, **kwargs):
-        was_trace_enabled = _is_enabled()
-        if was_trace_enabled:
-            disable()
-            try:
+        _is_func_called = False
+        try:
```

Comment:
note: Adding extra try-catch to absorb exceptions from `_is_enabled()` and `disable()` calls.

Reply:
Can we add a logger.debug for the case it fails on non-function call?